proto/gossip_delegates.go

```diff
@@ -323,3 +412,51 @@ func (gd *GossipDelegate) handleStateEvents() {
 		gd.UpdateSelfStatus(gd.currentState.NodeStatus())
 	}
 }
+
+func (gd *GossipDelegate) nodeNameToProbationID(nodeName string) string {
+	return "gossip-" + nodeName
+}
+
+func (gd *GossipDelegate) probationIDToNodeName(probationID string) string {
+	return strings.TrimPrefix(probationID, "gossip-")
+}
+
+// isNodeSuspect returns a boolean indicating whether a peer node should be put
+// in suspected Offline state. For the given nodeId, it finds out all its peers from
+// the same failure domain. It then pings all these nodes, and if even one of the pings
+// fails it assumes that the failure domain itself could be down and marks the input
+// node as suspected offline
```

Comment:
Behavior should be: If even one of pings succeeds the MetroDomain is alive

Reply:
done